drop-in/src/main/java/com/adyen/checkout/dropin/internal/provider/ComponentParsingProvider.kt

```diff
@@ -1,948 +0,0 @@
-        }
-
-        checkCompileOnly { CardComponent.PROVIDER.isPaymentMethodSupported(storedPaymentMethod) } -> {
-            val cardConfig: CardConfiguration =
-                getConfigurationForPaymentMethod(storedPaymentMethod, dropInConfiguration)
-            CardComponentProvider(dropInParams, sessionParams, analyticsRepository).get(
-                fragment = fragment,
-                storedPaymentMethod = storedPaymentMethod,
-                configuration = cardConfig,
-                callback = componentCallback as ComponentCallback<CardComponentState>,
-                key = storedPaymentMethod.id
-            )
-        }
-
-        checkCompileOnly { CashAppPayComponent.PROVIDER.isPaymentMethodSupported(storedPaymentMethod) } -> {
-            val cashAppPayConfig: CashAppPayConfiguration =
-                getConfigurationForPaymentMethod(storedPaymentMethod, dropInConfiguration)
-            CashAppPayComponentProvider(dropInParams, sessionParams).get(
-                fragment = fragment,
-                storedPaymentMethod = storedPaymentMethod,
-                configuration = cashAppPayConfig,
-                callback = componentCallback as ComponentCallback<CashAppPayComponentState>,
-                key = storedPaymentMethod.id
-            )
-        }
-
-        checkCompileOnly { BlikComponent.PROVIDER.isPaymentMethodSupported(storedPaymentMethod) } -> {
-            val blikConfig: BlikConfiguration =
-                getConfigurationForPaymentMethod(storedPaymentMethod, dropInConfiguration)
-            BlikComponentProvider(dropInParams, sessionParams, analyticsRepository).get(
-                fragment = fragment,
-                storedPaymentMethod = storedPaymentMethod,
-                configuration = blikConfig,
-                callback = componentCallback as ComponentCallback<BlikComponentState>,
-                key = storedPaymentMethod.id
-            )
-        }
-
-        else -> {
-            throw CheckoutException("Unable to find stored component for type - ${storedPaymentMethod.type}")
-        }
-    }.apply {
-        setOnRedirectListener(onRedirect)
-    }
-}
-
-/**
- * Provides a [PaymentComponent] from a [PaymentComponentProvider] using the [PaymentMethod] reference.
- *
- * @param fragment The Fragment which the PaymentComponent lifecycle will be bound to.
- * @param paymentMethod The payment method to be parsed.
- * @throws CheckoutException In case a component cannot be created.
- */
-@Suppress("LongMethod", "UNCHECKED_CAST", "LongParameterList", "CyclomaticComplexMethod")
-internal fun getComponentFor(
-    fragment: Fragment,
-    paymentMethod: PaymentMethod,
-    dropInConfiguration: DropInConfiguration,
-    amount: Amount?,
-    componentCallback: ComponentCallback<*>,
-    sessionDetails: SessionDetails?,
-    analyticsRepository: AnalyticsRepository,
-    onRedirect: () -> Unit,
-): PaymentComponent {
-    val dropInParams = dropInConfiguration.mapToParams(amount)
```

Comment:
This logic (passing the amount) was used to ensure the amount is updated in the partial payments flow. I cannot find the same logic in the new implementation so I think partial payments with the advanced flow might not be showing the right amount

Reply:
Good catch! Let me check this